ext/curl/interface.c

```diff
@@ -2106,16 +2106,169 @@ PHP_FUNCTION(curl_copy_handle)
 }
 /* }}} */
 
+static inline
+void php_curl_setopt_type_failure(zend_uchar exp, zval *arg, zend_bool nullable) { /* {{{ */
+	const char *exp_of = "be of type ";
+	const char *exp_type = "";
+
+	if (IS_CALLABLE == exp) {
+		exp_of = "be callable";
+	} else if (IS_ITERABLE == exp) {
+		exp_of = "be iterable";
+	} else {
+		exp_type = zend_get_type_by_const(exp);
+	}
+
+	zend_type_error("Argument 3 passed to curl_setopt() must %s%s, %s given",
+	                exp_of, exp_type, zend_zval_type_name(arg));
```

Comment:
I agree.  I even started to write it that way initially, but then I realize there was no reverse map of option values to option names.  Since that point I added the header map so it could be added back in, but the whole thing is pretty much been shot down, so the whole thing is moot.